### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by Alteqa**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -196,10 +196,47 @@ The sequence diagram below closely describes the interaction between the various
...
+<img src="images/EditPersonActivityDiagram.png" width="800" />
+
+* Step 4. The `parse` command in `editCommandParser` return an instance of `editCommand`.
+* Step 5. The `LogicManager` calls the `execute` method in `editCommand`.
+* Step 6. The `execute` method in `editCommand` executes and creates a new edited person.
+* Step 6. The `execute` method in `editCommand` calls the `setPerson` in model to update the details of the respective person.
+* Step 7. The `execute` method in `editCommand` also iterates through the `ObservableList<Appointments>` and retrieves all appointments that have the person to be edited, and calls the `setDoctorNric` or `setPatientNric` methods to update all relevant appointments related to the patient or doctor.
+* Step 8. Success message gets printed onto the results display to notify user.
+
+The sequence diagram below closely describes the interaction between the various components during the execution of the `DeleteCommand`.
```
Comment (Kappaccinoh) [inline docs/DeveloperGuide.md:223]: change to editCommand

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -284,6 +321,44 @@ The sequence diagram below closely describes the interaction between the various
 * Furthermore, it might get confusing for the user if everything was dumped into the same list of them to sieve through. Perhaps the user was only concerned with looking up patients in which case the appointments would simple be added clutter.
 * The increased level of integration would also be problems for implementation and testing as existing functionality would have to be adapated exposing the system to more risks and potential for bugs. Eg: the classes would have to change from `Person` to `Entry` in a number of different places.
 
+
+### Edit `Appointment`
+Edits an `Appointment` entry by indicating their `Index`.
+This command is implemented through the `EditAppointmentCommand` class which extend the `Command` class.
```
Comment (Kappaccinoh) [inline docs/DeveloperGuide.md:327]: extends

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -284,6 +321,44 @@ The sequence diagram below closely describes the interaction between the various
...
+* Step 3. The `parse` command in `editAppointmentCommandParser` calls `ParserUtil` to create instances of objects for each of the fields.
+  * If there are any missing fields, a `CommandException` is thrown.
+  * If input arguments does not match contraints for the fields, a `IllegalArgumentException` is thrown.
+  * If the provided `index` is invalid, a `CommandException` is thrown.
+
+The activity diagram below demonstrates this error handling process in more detail.
+
+<img src="images/EditAppointmentActivityDiagram.png" width="800" />
+
+* Step 4. The `parse` command in `editAppointmentCommandParser` return an instance of `editAppointmentCommand`.
```
Comment (Kappaccinoh) [inline docs/DeveloperGuide.md:340]: returns

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -284,6 +321,44 @@ The sequence diagram below closely describes the interaction between the various
...
+* Step 5. The `LogicManager` calls the `execute` method in `editAppointmentCommand`.
+* Step 6. The `execute` method in `editAppointmentCommand` executes and calls `setAppointment` in model to set an updated appointment into the system.
+* Step 7. Success message gets printed onto the results display to notify user.
+
+The sequence diagram below closely describes the interaction between the various components during the execution of the `DeleteAppointmentCommand`.
```
Comment (Kappaccinoh) [inline docs/DeveloperGuide.md:345]: EditAppointmentCommand

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -284,6 +321,44 @@ The sequence diagram below closely describes the interaction between the various
...
+3. This parallelism is also reflected in the backend code, and hence is very similar to how editing a `Person` is implemented - this is mostly seen through the naming conventions of the classes related to `EditPerson`, such as `EditAppointment`
+4. This results in a more familiar experience for both users and developers alike as there is familiarity and some level of consistency when dealing with `Person` and `Appointment` classes.
+
+Alternative implementation for consideration
+1. Since both classes exhibit similarities in both code structure and behaviour, we might consider creating a generic class distinguished between `Person` and `Appointment` via enums to handle edits.
+2. This will centralise the behaviours, and reduce the amount of code needed to perform the edit function.
+   3. A further extension is to do so with all other overlapping functionalities, such as `add` or `delete`, however we leave that possibility for future discussion and refinement.
```
Comment (Kappaccinoh) [inline docs/DeveloperGuide.md:358]: indentation before 3.

**Review by alfaloo — APPROVED** [review]
Amazing

**Pull request merged**